ykrt/src/compile/jitc_llvm.rs

```diff
@@ -79,7 +79,7 @@ impl JITCLLVM {
         Arc::new(JITCLLVM)
     }
 
-    fn encode_trace(&self, irtrace: &Vec<TracedAOTBlock>) -> (Vec<*const i8>, Vec<usize>, usize) {
```

Comment:
`ProcessedItem` is such a generic name.

At least with `TracedAOTBlock` we know the granularity of what we are working with (blocks).

Honestly, I'd rather keep the old name and just know that when you trace an AOT block it can include promotions, although I'm guessing that you will object.

Reply:
OK, I see that later `ProcessedItem` will become an enum. I this case I agree that `TracedAOTBlock` is no longer a good name, but I still think `ProcessedItem` is not a good name.

How about one of the following:
 - `AOTObservation`
 - `AOTTraceEvent`
 - `AOTEvent`

Reply:
Or maybe call the enum `TraceAction` or similar, and give the variants verb names, like `ExecuteBlock` and `ObservePromotion`?

Reply:
The enum can't have "AOT" in the name. `TraceAction` could work with `Block` and `Promotion`? [`Execute` and in particular `Observe` seem like padding to me.]

Reply:
I think that would be more descriptive than `ProcessedItem`, so if you are happy, so am I :)

Reply:
https://github.com/ykjit/yk/pull/978/commits/a92f1feed1fad3649533d4db54cec703ba50589a changes the enum name (I think the enum variant names are probably OK for now as-is). If you do like this, it'll have to stay as a separate commit as rebasing this earlier will be a nightmare. [In other words: this PR could be merged as-is if -- and only if! -- you like the PR.]